src/components/swap/SwapBox.vue

```diff
@@ -183,6 +214,150 @@ export default defineComponent({
+      }
+    },
+    async onSwap() {
+      if (!this.swapFrom.value) {
+        console.error("You need to estimate a swap amount!");
+        return;
+      }
+
+      this.showSpinner = true;
+
+      const estimatedGasFee = await fetch(
+        `${process.env.VUE_APP_RELAYER_ENDPOINT}/estimated-gas`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            amount: this.swapFrom.value,
+            unitType: "wei",
+          }),
+        }
+      );
+
+      const signedData = await this.signWithMetamask(estimatedGasFee); // needs to match what is in contract
+      // calll swap
+      if (!signedData) {
+        // todo: display error
+        console.error("no signed data!");
+        return null;
+      }
+
+      try {
+        const response = await fetch(
+          // 0xCa1D8715989D70173DdaE7e7b2111f49dE88d1c4
+          `${process.env.VUE_APP_RELAYER_ENDPOINT}/swap`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              swapData: signedData.message.swapData,
+              deadline: signedData.message.deadline,
+              relayerAddress: process.env.VUE_APP_RELAYER_ADDRESS,
+              v: signedData.v,
+              r: signedData.r,
+              s: signedData.s,
+              estimatedGasFee,
+              sideTokenBtcContractAddress: this.swapFrom.address,
+            }),
+          }
+        );
+      } catch (err) {
+        console.error("couldnt swap", err);
+      } finally {
+        this.showSpinner = false;
+      }
+    },
+    async signWithMetamask(estimatedGas) {
+      try {
+        return this.signMessage([
+          this.account,
+          JSON.stringify(this.parseMessageToSign(estimatedGas)),
+        ]);
+      } catch (err) {
+        console.error("couldnt sign message", err);
+      }
+      return null;
+    },
+    signMessage(params) {
+      return new Promise((resolve, reject) => {
+        this.web3.currentProvider.sendAsync(
+          {
+            method: "eth_signTypedData_v4",
+            params,
+            from: this.account,
+          },
+          (err, result) => {
+            if (err) reject(err);
+            if (result?.error) reject("result with error ", result?.error);
+            if (!result.result) reject(new Error("Empty or undefined result!"));
+
+            const signature = result.result.substring(2);
+            const r = `0x${signature.substring(0, 64)}`;
+            const s = `0x${signature.substring(64, 128)}`;
+            const v = parseInt(signature.substring(128, 130), 16);
+
+            resolve({ r, s, v });
+          }
+        );
+      });
+    },
+    // parseMessageToSign(nonce) {
+    parseMessageToSign(estimatedGas) {
+      return {
+        domain: {
+          name: "RSK Relayer",
```

Comment:
this needs to be adapted to the latest version of the typed data (check tokenbridge-backend script)